core/src/main/java/com/linecorp/armeria/client/encoding/GzipStreamDecoderFactory.java

```diff
@@ -22,7 +22,11 @@
 /**
  * A {@link StreamDecoderFactory} which supports the 'gzip' encoding.
  */
-public class GzipStreamDecoderFactory implements StreamDecoderFactory {
+final class GzipStreamDecoderFactory implements StreamDecoderFactory {
+
+    static final GzipStreamDecoderFactory INSTANCE = new GzipStreamDecoderFactory();
```

Comment:
Maybe use enum singleton pattern for simple singletons like these

Reply:
Good idea. Merged the two classes into one enum class.